src/ecoscope-workflows-core/ecoscope_workflows_core/indexes.py

```diff
@@ -67,6 +67,13 @@
 ]
 
 
+TemporalIndexSelector = tuple(value for value in strftime_directives.values())
+temporal_index_json_schema_extra = oneOf(
+    const=[k for k in strftime_directives.keys()],
+    title=[v for v in strftime_directives.values()],
+).model_dump()
```

Comment:
```suggestion
temporal_index_json_schema_extra = {
    "oneOf": [
         oneOf(const=k, title=v).model_dump()
         for k, v in strftime_directives
    ]
}
```

oops i lead you astray with this recommendation, it should be something like this

Reply:
no stress! on it :smile:

Reply:
Local recompile definitely seems quicker for iterating on things like this

Reply:
fair!